src/db/tileDatabase.ts

```diff
@@ -0,0 +1,28 @@
+import * as sqlite3 from "sqlite3";
+import { TilesRow } from "../types/db";
+
+export class TileDatabase {
+    db: sqlite3.Database;
+
+    constructor(dbFileName: string) {
+        this.db = new sqlite3.Database(dbFileName, sqlite3.OPEN_READONLY);
+    }
+
+    getTileData(z: number, x: number, y: number): object {
+        return new Promise((resolve, reject) => {
+            this.db.get(
+                `SELECT tile_data FROM tiles WHERE zoom_level = ${z} AND tile_column = ${x} AND tile_row = ${y}`,
```

Comment:
Wondering if this is vulnerable to SQL injection (particularly `y` since it could be `; DROP TABLE important_table`). If the `get` function doesn't already take care of it we should use SQL `?` parameters.

Reply:
Fair point! The next PR returns a 400 if we get non-integer z, x or y parameters so injection shouldn't be an issue. I'm not sure if sqlite3 supports variables but will check and add in if so.

Reply:
I'm going to update this to use the `node-sqlite` package, which builds on `sqlite3` - this supports sql parameters, and also generics to specify row type so we can be a little less mysterious here.